### ISSUE TITLE
[BUG] renaming devices causes out of sync

### ISSUE THREAD

**Issue by philuxe** [yllibed/Zigbee2MqttAssistant#130]
I have renamed two of my devices using the assistant, then restarted the computer hosting the docker container. After the host reboot renamed devices are duplicated. Those that have been renamed are showing "NO COMPONENTS Maybe this device is not (yet) supported by Zigbee2Mqtt?" Some new devices entries are shown with the original friendly name (like before renaming)

## To Reproduce

1. rename some devices
2. reboot the docker host
3. see duplicated devices

## Installation
 - Version of Zigbee2Mqtt: 1.7.1
 - Version of Zigbee2MqttAssistant: 0.3.105 (0.3.105+Branch.master.Sha.b1c45d4ca947b2701c12bdf7d0e8be4083a0a2fe) 
 - Installation type (HASS.IO, Docker...): Docker

![Screenshot_1](https://user-images.githubusercontent.com/16466030/69827923-dbd6ef00-1219-11ea-9bdd-57e94d1adb0b.png)

**Comment by titomontes** [comment]
Same issue here.

**Comment by carldebilly** [comment]
Since Zigbee2Mqtt won't publish new _Home Assistant Discovery topics_ for the renamed device.  You need to reboot Z2M for that (I think this problem was never reports to Z2M).

Since Z2MA is using HA discovery topics for components, it's creating a _fantom_ device with the old name.

The best way to fix this is to reboot Z2M.

**Comment by GrumpyMeow** [comment]
I'm also having this issue. I'm pretty sure that i did restart the Zigbee2Mqtt container after renaming the devices. I did this last night. Today during the day the fantom devices got added.
I do see a pattern in that i renamed these devices and restarted the Zigbee2Mqtt & Zigbee2MqttAssistant containers after initiating the rename. I did it this way as i kept having trouble adding these devices to Z2M.
Can it be that the rename-action didn't got completely written to disk due to caching for example?

**Comment by carldebilly** [comment]
I think it's more a Z2M bug :-)

**Comment by GrumpyMeow** [comment]
Yes, you're right.. These are indeed Z2M configuration files. Thus it probably is more a Z2M bug.

**Comment by CodeFinder2** [comment]
I have the same problem and I doubt that it's a Z2M bug. :see_no_evil: I've restarted Z2M and Z2MA and the fantom devices stay in my list: :thinking: 
![Screenshot_20200123-002520](https://user-images.githubusercontent.com/20599588/72943781-3bafc900-3d77-11ea-9aa7-2f2545656c88.jpg)

Even if it's a Z2M bug, an issue should be opened for it there...

@GrumpyMeow would you mind sharing your config files (configuration.yaml, state.json, database.db, possibly devices.yaml) here or at least the relevant parts of them indicating the out-of-sync state? I checked mine and they seem to be fine/sync.

**Comment by CodeFinder2** [comment]
Can someone having this issue try to reboot the host to see if it gets rid of fantom devices?

**Comment by ljpaff** [comment]
I have rebooted Mosquitto Server and Z2MA but having the same phantom devices...

**Comment by CodeFinder2** [comment]
Yeah, I also rebooted virtually everything but it happens again. They appear and go away for no reason but most of the time, there are phantom devices listed. 😕

**Comment by shirou93** [comment]
same issue

**Comment by SPEC1AL1ST** [comment]
Same issue!

**Comment by shirou93** [comment]
I delete old devices topics and restart z2massistant, Blank devices disappear.
